tool/tsh/common/tsh.go

```diff
@@ -2546,6 +2546,10 @@ func createAccessRequest(cf *CLIConf) (types.AccessRequest, error) {
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+
+	// prevent the same resource from being repeated in the request
+	cf.RequestedResourceIDs = apiutils.Deduplicate(cf.RequestedResourceIDs)
```

Comment:
I don't think we have lots of well-meaning people putting tons of duplicate request IDs into the tsh CLI. If we want to prevent someone breaking request listing with tons of (duplicated) request IDs I think we should do it on the backend somewhere during request validation, probably right before this line https://github.com/gravitational/teleport/blob/f75e8fc3bc8172c921bdb226cc50c0e4a1f67b6a/lib/services/access_request.go#L1562

and nit: we could test it too

Reply:
actually, right before this line https://github.com/gravitational/teleport/blob/9a0f8d6f14da3029c2c69440fa07619a6c9bb48a/lib/services/access_request.go#L1557

Reply:
Should we do it in both places?

Deduplicating a resource ID on the backend is a bit trickier, as it's a protobuf generated type with hidden fields and not a simple string. Additionally, if we only do it on the backend then that means the backend has to at least be able to receive the gRPC, which could leave a confusing error message if the request is too large.

Reply:
we could do both, but I just don't really believe that any well-meaning person is sending a huge list of duplicated resource IDs on the tsh CLI, if a security researcher does it i don't think the grpc message size error matters

if we want to prevent "the size of the resource [exceeding] the limits of a gRPC message and break[ing] listing" the only effective thing to do is enforce a limit on the backend, and maybe deduplicate